src/libcore/iter/range.rs

```diff
@@ -353,6 +304,23 @@ impl<A: Step> Iterator for ops::RangeFrom<A> where
     fn size_hint(&self) -> (usize, Option<usize>) {
         (usize::MAX, None)
     }
+
+    #[inline]
+    fn count(self) -> usize {
+        usize::MAX
```

Comment:
@scottmcm the documentation of `Iterator::count` says:

> The method does no guarding against overflows, so counting elements of an iterator with more than usize::MAX elements either produces the wrong result or panics. If debug assertions are enabled, a panic is guaranteed.

so yes, it should panic, as you say.